ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -53,30 +53,56 @@ let sync_required ~__context ~host =
     end
   | _ -> raise Api_errors.(Server_error (internal_error, ["Cannot have more than one Cluster object per pool currently"]))
 
+let call_api_function_with_alert ~__context ~msg ~cls ~obj_uuid ~body
+  ~(api_func : (Rpc.call -> Rpc.response) -> API.ref_session -> unit) =
+    Helpers.call_api_functions ~__context
+      (fun rpc session_id ->
+        try
+          api_func rpc session_id
+        with err ->
+          let body = Printf.sprintf "Error: %s\nMessage: %s" ExnHelper.(string_of_exn err) body in
```

Comment:
Not sure how XenCenter handles messages with newlines, can you try to see if it looks alright?

Reply:
Apparently messages can only be one line, so I don't think I can include the error.